x-pack/plugins/reporting/server/lib/screenshots/observable.ts

```diff
@@ -60,23 +61,33 @@ export function screenshotsObservableFactory(
         apmCreatePage?.end();
         exit$.subscribe({ error: () => apmTrans?.end() });
 
-        return Rx.from(urls).pipe(
-          concatMap((url, index) => {
+        return Rx.from(urlsOrUrlLocatorTuples).pipe(
+          concatMap((urlOrUrlLocatorTuple, index) => {
             const setup$: Rx.Observable<ScreenSetupData> = Rx.of(1).pipe(
               mergeMap(() => {
+                let url: string;
+                let locator: undefined | LocatorParams;
+
+                if (typeof urlOrUrlLocatorTuple === 'string') {
+                  url = urlOrUrlLocatorTuple;
+                } else {
+                  [url, locator] = urlOrUrlLocatorTuple;
+                }
+
                 // If we're moving to another page in the app, we'll want to wait for the app to tell us
                 // it's loaded the next page.
-                const page = index + 1;
+                const p = index + 1;
                 const pageLoadSelector =
-                  page > 1 ? `[data-shared-page="${page}"]` : DEFAULT_PAGELOAD_SELECTOR;
+                  p > 1 ? `[data-shared-page="${p}"]` : DEFAULT_PAGELOAD_SELECTOR;
```

Comment:
Perhaps all of the above code in this function could just be part of the `openUrl` function?